mbus/mbus-tcp.c

```diff
@@ -52,34 +49,34 @@ mbus_tcp_connect(mbus_handle *handle)
         return -1;
 
     host = tcp_data->host;
-    port = tcp_data->port;
+    sprintf(port, "%d", tcp_data->port);
 
-    //
-    // create the TCP connection
-    //
-    if ((handle->fd = socket(AF_INET,SOCK_STREAM, 0)) < 0)
-    {
-        snprintf(error_str, sizeof(error_str), "%s: failed to setup a socket.", __PRETTY_FUNCTION__);
+    memset(&hints, 0, sizeof hints);
+    hints.ai_family = AF_UNSPEC;
+    hints.ai_socktype = SOCK_STREAM;
+
+    if ((status = getaddrinfo(host, port, &hints, &servinfo)) != 0) {
+        snprintf(error_str, sizeof(error_str), "%s: getaddrinfo: %s", __PRETTY_FUNCTION__, gai_strerror(status));
         mbus_error_str_set(error_str);
         return -1;
     }
 
-    s.sin_family = AF_INET;
-    s.sin_port = htons(port);
+    // loop through all the results and connect to the first we can
+    for(p = servinfo; p != NULL; p = p->ai_next) {
```

Comment:
Please add a space between for and (

Reply:
Sorry, typo, long time to do something in C, mostly using python for now